accountsynchr/trumba_gws.py

```diff
@@ -1,175 +1,124 @@
+        self.ttl_showon_grp_synced += 1
+        if not is_new_calendar:
+            self.sync_showon_perms(trumba_cal, uwcal_group)
+
+    def sync_edit_perms(self, trumba_cal, uwcal_group):
+        """
+        Delete the edit permission no longer having the membership
+        """
+        for perm in get_cal_permissions(trumba_cal):
+            if not perm.in_editor_group():
+                continue
+            uwnetid = perm.uwnetid
+            if not_member(uwcal_group, uwnetid):
+                if remove_permission(trumba_cal, uwnetid) is True:
+                    self.del_editor_perm_counts += 1
+                    logger.info("Removed editor {0} from {1}".format(
+                            uwnetid, trumba_cal.name))
+                else:
+                    self.append_error(
+                        "Failed to remove editor {0} from {1}\n".format(
+                            uwnetid, trumba_cal.name))
 
-    def sync_showon_perms(self, trumba_cal, showon_group):
+    def sync_showon_perms(self, trumba_cal, uwcal_group):
         """
         Delete the showon permission no longer having the correspoding
-        member in the showon group
-        """
-        showon_perm_list = self.per_m.get_showon_permissions(trumba_cal)
-        for perm in showon_perm_list:
-            # currently having showon permission in Trumba
-            if not self.mem_m.is_member(showon_group.name, perm.uwnetid):
-                # not an showon group member
-
-                logger.info("TO REMOVE showon %s from %s" % (perm.uwnetid,
-                                                             trumba_cal))
-                success = self.per_m.remove_permission(trumba_cal,
-                                                       perm.uwnetid)
-                if success:
-                    self.del_showon_perm_counts =\
-                        self.del_showon_perm_counts + 1
+        membership
+        """
+        for perm in get_cal_permissions(trumba_cal):
+            if not perm.in_showon_group():
+                continue
+            uwnetid = perm.uwnetid
+            if not_member(uwcal_group, uwnetid):
+                if remove_permission(trumba_cal, uwnetid) is True:
+                    logger.info("Removed showon {0} from {1}".format(
+                            uwnetid, trumba_cal.name))
+                    self.del_showon_perm_counts += 1
                 else:
-                    logger.error(
-                        "Failed to remove showon %s from %s" % (perm.uwnetid,
-                                                                trumba_cal))
-                    self.err_del_showon_perm_counts =\
-                        self.err_del_showon_perm_counts + 1
+                    self.append_error(
+                        "Failed to remove showon {0} from {1}\n".format(
+                            uwnetid, trumba_cal.name))
+
+
+def not_member(uwcal_group, uwnetid):
+    for gmember in uwcal_group.members:
```

Comment:
Is it not feasible to use the `in` keyword with a custom __eq__ method here?

Reply:
we check  gmember.name == uwnetid.  it will require adding a method in uw_gws model